test/Tests/src/Semantic/Instrumentor_Spec.enso

```diff
@@ -0,0 +1,44 @@
+from Standard.Base import all
+from Standard.Test import all
+
+fib n = if n <= 1 then 1 else
+    a = fib n-1
+    b = fib n-2
+    a+b
+
+spec =
+    Test.group "Instrument fibonacci" <|
+        Test.specify "collect and filter on return updates" <|
+            b = Vector.new_builder
+
+            collect uuid:Text result =
+                a_plus_b_uuid = "00000000-aaaa-bbbb-0000-000000000000" # UUID for a+b
+                if uuid == a_plus_b_uuid then
+                    b.append result
+                Nothing
+
+            Meta.meta .fib . instrument . on_return collect . activate
+            result = fib 10
+
+            v = b.to_vector
+
+            v.length . should_equal 1
+            v.at 0 . should_equal 89
+            result . should_equal 89
+
+        Test.specify "replay with caches and specify different result" <|
+            replay uuid:Text = case uuid of
+                "00000000-aaaa-bbbb-0000-000000000000" -> 42
+                _ -> Nothing
+
+            Meta.meta .fib . instrument . on_enter replay . activate
+            result = fib 10
+
+            result . should_equal 42
```

Comment:
This test is awesome, but for completeness I think we should also have a test with `on_function` and `on_exception`.

Reply:
`on_exception` removed. `on_call` (originally `on_function`) is tested since bfa4577]